src/promptflow/promptflow/_core/operation_context.py

```diff
@@ -111,14 +111,38 @@ def get_user_agent(self):
 
         def parts():
             # Be careful not to use "self.user_agent" as it will recursively call get_user_agent in __getattr__
-            agent = self.get(self._USER_AGENT, '')
+            agent = self.get(self._USER_AGENT, "")
             yield agent
             promptflow_agent = f"promptflow/{VERSION}"
-            yield promptflow_agent if promptflow_agent not in agent else ''
-            user_agent = os.environ.get(USER_AGENT, '').strip()
-            yield user_agent if user_agent not in agent else ''
-            pf_user_agent = os.environ.get(PF_USER_AGENT, '').strip()
-            yield pf_user_agent if pf_user_agent not in agent else ''
+            yield promptflow_agent if promptflow_agent not in agent else ""
+            user_agent = os.environ.get(USER_AGENT, "").strip()
+            yield user_agent if user_agent not in agent else ""
+            pf_user_agent = os.environ.get(PF_USER_AGENT, "").strip()
+            yield pf_user_agent if pf_user_agent not in agent else ""
+
+        # strip to avoid leading or trailing spaces, which may cause error when sending request
+        ua = " ".join(parts()).strip()
+        return ua
+
+    def get_client_user_agent(self):
```

Comment:
Looks like it is a telemetry specific logic not a common logic for operation context

Reply:
Please add @liucheng-ms or me for operation context related changes to avoid unexpected changes in telemetry.

Reply:
Moved this logic to separate file.